lib/main.js

```diff
@@ -172,83 +164,93 @@ module.exports = class Router extends EventEmitter {
+                    type,
+                });
+                this.emit(
+                    'request success',
+                    res.locals.track,
+                    req.method,
+                    req.path,
+                    response.file
+                );
+                res.json(response);
+            } catch (e) {
+                this.logger.error('Failed creating bundle', e);
+                if (e.message.includes('No file with name')) {
                     return next(
-                        boom.boomify(error, {
-                            statusCode: 400,
-                            message: 'Invalid POST-body',
+                        boom.boomify(e, {
+                            statusCode: 404,
                         })
                     );
                 }
-                res.locals.payload = payload;
-                next();
-            });
+                next(e);
+            }
         };
     }
 
-    postBundlePersistCallback() {
-        return (req, res, next) => {
-            let uri = this.buildUri('bundle', req.headers.host, req.secure);
-
-            const fileStreams = res.locals.payload.map(file =>
-                this.sink.reader(file)
-            );
-
-            let fileWriteStream;
-            let reader;
-            if (req.handleAs === 'css') {
-                fileWriteStream = this.sink.writer('css');
-                reader = new CssReader(fileStreams);
-            } else {
-                fileWriteStream = this.sink.writer('js');
-                reader = new JsReader(fileStreams);
-            }
-
-            fileWriteStream.on('file saved', (id, file) => {
-                uri += file;
-                res.locals.response = {
-                    file,
-                    uri,
-                };
-                next();
-            });
-
-            fileWriteStream.on(
-                'file not saved',
-                this.onError(next, 'Generated file could not be saved')
-            );
-
-            fileWriteStream.on('error', this.onError(next));
+    async bundle({ type, feedIds, uri }) {
+        assert(
+            Array.isArray(feedIds) && feedIds.length > 0,
+            `Expected at least 1 feed id, but got ${feedIds}`
+        );
+        this.logger.info(`Processing feeds ${feedIds.join(', ')}`);
+        const feeds = await Promise.all(
+            feedIds.map(file =>
+                this.sink
+                    .get(file)
+                    .then(content => JSON.parse(content))
+                    .then(feed => {
+                        assert(
+                            Array.isArray(feed),
+                            `Expected feed entry to be an array, but got ${typeof feed}`
+                        );
+                        return feed;
+                    })
+            )
+        );
 
-            reader.on('error', this.onError(next));
+        let bundle;
+        if (type === 'css') {
```

Comment:
should we be explicit with `js` and throw if not one of those 2?

Reply:
it will throw higher up start-up-time.